source/obstacle/Obstacle.hx

```diff
@@ -1,57 +1,81 @@
 package obstacle;
 
+import flixel.FlxBasic;
 import flixel.FlxG;
 import flixel.FlxSprite;
 
-class Obstacle extends FlxSprite
-{
-	public function new(?parameters:ObstacleParameters)
-	{
+class Obstacle extends FlxSprite {
+	public function new(?parameters:ObstacleParameters) {
 		super();
 
 		immovable = true;
 
-		if (parameters != null)
-		{
+		if (parameters != null) {
+			// this gives the obstacle its init values
 			init(parameters);
 		}
 	}
 
-	public function init(parameters:ObstacleParameters)
-	{
+	public function init(parameters:ObstacleParameters) {
+		// revive the obstace to be drawn again
 		super.reset(parameters.x, parameters.y);
 
+		// set the velocity to whatever the speed is but neg
+		var upOrDown = FlxG.random.int(0, 1);
 		velocity.x = -parameters.speed;
 
+		if (upOrDown == 1) {
+			velocity.y = -(FlxG.random.float(parameters.speed / 2, parameters.speed));
+		} else {
+			velocity.y = (FlxG.random.float(parameters.speed / 2, parameters.speed));
+		}
+
+		// used to create as a rectangle
 		makeGraphic(parameters.width, parameters.height);
 	}
 
-	override public function update(elapsed:Float)
-	{
-		if (movedOffScreen())
-		{
+	override public function update(elapsed:Float) {
+		// if the object has moved off screen
+		if (movedOffScreen()) {
+			// dont draw it
 			kill();
 		}
 
+		// adds the bounce effect
+		/* if (y < 0 || y > FlxG.height - height) {
+			velocity.y = -velocity.y;
+		}*/
```

Comment:
If this is going to stay commented, I think we should remove it entirely instead. If you want to keep it around to give yourself the option of using it later, let's put it in a private method that doesn't get called.